Cereal/CerealDecoder.swift

```diff
@@ -238,14 +275,14 @@ public struct CerealDecoder {
     - returns:      The instantiated object, or nil if no object was at the specified key.
     */
     public func decodeCereal<DecodedKeyType: protocol<CerealType, Hashable>, DecodedValueType: CerealRepresentable>(key: String) throws -> [[DecodedKeyType: DecodedValueType]]? {
-        guard let data = items[key] else {
+        guard let data = self.itemForKey(key), case let .ArrayValue(items) = data else {
```

Comment:
Hmm, how would you feel if the data doesn't match the expected type we throw an error instead of returning nil?